openc3/lib/openc3/utilities/store_queued.rb

```diff
@@ -0,0 +1,113 @@
+# encoding: ascii-8bit
+
+# Copyright 2024 OpenC3, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+require 'openc3/utilities/store'
+
+module OpenC3
+  class StoreQueued
+    # How often in seconds the store is updated
+    UPDATE_INTERVAL = 2
+
+    # Variable that holds the singleton instance
+    @instance = nil
+
+    # Mutex used to ensure that only one instance is created
+    @@instance_mutex = Mutex.new
+
+    # Thread used to post metrics across all classes
+    @@update_thread = nil
+
+    # Sleeper used to delay update thread
+    @@update_sleeper = nil
+
+    # Get the singleton instance
+    def self.instance()
+      return @instance if @instance
+
+      @@instance_mutex.synchronize do
+        @instance ||= self.new()
+        return @instance
+      end
+    end
+
+    def self.shutdown
```

Comment:
who calls this?

Reply:
no one ... it's left over from the metrics thread I borrowed

Reply:
Should be added to at_exit when instance created

Reply:
Should be instance method